### PR TITLE
Make categorize.py aware of missing categories

### PR DESCRIPTION
**Pull request by vmoens**
Proposes changes in `categorize.py` to include categories (release notes tags) that occurred since the last release but were not listed in `common.categories`.

Example:
First uncategorized commit without these changes:
```
Test Plan: CI

Differential Revision: D54791087

Pull Request resolved: https://github.com/pytorch/pytorch/pull/121712
Approved by: https://github.com/ydwu4

Files changed: ['test/export/test_torchbind.py', 'torch/_ops.py']

Labels: ['fb-exported', 'Merged', 'ciflow/trunk']

Current category: Uncategorized

Select from: Uncategorized, lazy, hub, mobile, jit, visualization, onnx, caffe2, amd, rocm, cuda, cpu, cudnn, xla, benchmark, profiler, performance_as_product, package, dispatcher, releng, fx, code_coverage, vulkan, skip, composability, mps, intel, functorch, gnn, distributions, serialization, meta_frontend, nn_frontend, linalg_frontend, cpp_frontend, python_frontend, complex_frontend, vmap_frontend, autograd_frontend, build_frontend, memory_format_frontend, foreach_frontend, dataloader_frontend, sparse_frontend, nested tensor_frontend, optimizer_frontend, dynamo, inductor, quantization, distributed
```

With these changes:
```
Test Plan: CI

Differential Revision: D54791087

Pull Request resolved: https://github.com/pytorch/pytorch/pull/121712
Approved by: https://github.com/ydwu4

Files changed: ['test/export/test_torchbind.py', 'torch/_ops.py']

Labels: ['fb-exported', 'Merged', 'ciflow/trunk']

Current category: Uncategorized

Select from: Uncategorized, lazy, hub, mobile, jit, visualization, onnx, caffe2, amd, rocm, cuda, cpu, cudnn, xla, benchmark, profiler, performance_as_product, package, dispatcher, releng, fx, code_coverage, vulkan, skip, composability, mps, intel, functorch, gnn, distributions, serialization, meta_frontend, nn_frontend, linalg_frontend, cpp_frontend, python_frontend, complex_frontend, vmap_frontend, autograd_frontend, build_frontend, memory_format_frontend, foreach_frontend, dataloader_frontend, sparse_frontend, nested tensor_frontend, optimizer_frontend, dynamo, inductor, quantization, distributed, distributed (checkpoint), package/deploy, distributed (pipeline), distributed (dtensor), distributed (tools), xpu, export, distributed (fsdp2), distributed (torchelastic), optim
```

### CODE REVIEW

**Review by atalman — APPROVED** [review]
lgtm

File: scripts/release_notes/categorize.py [hunk 1]
```diff
@@ -48,7 +48,11 @@ def __init__(self, path, category="Uncategorized", use_classifier: bool = False)
         self.category = category
 
     def categorize(self):
+        existing_categories = {commit.category for commit in self.commits.commits}
+        common.categories += existing_categories - set(common.categories)
+
         commits = self.commits.filter(category=self.category)
+        # raise RuntimeError(f'categories {set(commit.category for commit in commits)}')
```
Comment (drisspg) [inline scripts/release_notes/categorize.py:55]: do you want this here?

**Comment by drisspg** [comment]
I think when I wrote some of this the thought would be that you would figure out what categories are missing and actually add them or merge them into another. Its hard to keep track of all the labels though but in theory I though they should be in a list somewhere

**Comment by vmoens** [comment]
@drisspg I guess here my take was that we may add or remove lablels, and the PoC for the RNs is likely someone unfamiliar with these lablels. 
Having the list automatically completed based on the labels that exist would help that person with a minimum head-scratching: otherwise, we should either (a) add the lablels to the list (each RNs PoC must then make a PR to update the list manually) or (b) stick to the list but important labels are missing (eg, export) and this will be a recurrent issue. 

IMO labels should not be hard coded in the repo, it's going to be hard to keep track of them all long-term (and we'll end up with an ever increasing list with probably many dead labels in it).

**Pull request closed without merging**